### PR TITLE
Upgraded Next.js to 15.0.1

### LINKED ISSUE #597: Upgrade to Next.js 14+
## Why do we need this ?

This is a regular updates and hygiene thing.
Next.js 14 offers various performance and tooling improvements that we could benefit from.

The main thing to keep in mind for this one is that pulling this off will need thorough and comprehensive testing of the whole app, and builds.

Upgrading Next.js should include updating any related packages (such as React, etc.), and any resulting changes to types or how third party packages/libs/functions are used.

Basically, nothing should really much different for any dev or user when they use Middleware after this issue is resolved.

### PR DESCRIPTION
**Pull request by Reterics**
## Linked Issue(s)

Fixes #597

## Acceptance Criteria fulfillment

Viewed/Tested pages:

 - [x] /teams
    - [x] Add, remove, edit teams
 - [x] /dora-metrics
    - [x] Settings -> Configure Production branches popup
    - [x] Pull Request Insights
    - [x] Calendar date changing
 - [x] /integrations 
    - [x] Removed Token, configured again
 - [x] /settings
 - [x] /system-logs
 - [x] Opened Help Assistant

## Proposed changes (including videos or screenshots)

According to the article on https://nextjs.org/blog/next-15, Next.js 15 is officially stable and ready for production, so I made 2 commits in this PR: (_Reason: in case if we need Next.js 14 instead_)

 1. I updated to version 14, and did my testcases above
 2. and then updated to 15 and did them again.

image

## Further comments

UI related warnings were there before the update as well such as:  (There were not introduced in this PR)
 - _"CreateTeams.tsx:259 Warning: A props object containing a "key" prop is being spread into JSX:"_
 - _"DeploymentInsightsOverlay.tsx:364 Warning: Each child in a list should have a unique "key" prop."_

### CODE REVIEW

**Comment by jayantbh** [comment]
Well, that's a bigger PR than I was expecting. 😄 
Let's get this merged before hacktoberfest ends! 👌🏽 🚀

File: .github/workflows/pre-commit.yml
Comment (jayantbh) [inline .github/workflows/pre-commit.yml:1]: If you look at the latest version of contributing.md, you'll see that this stuff isn't there anymore, replaced by something else. Might need to rebase.

**Review by jayantbh — CHANGES_REQUESTED** [review]
Honestly the change I'm requesting at the moment isn't even worth blocking the PR over, except that the contributing guide changed significantly and this might need to weird confusions if this were to be merged as is.

Also surprised that there's not bigger breakages or type changes. We'll check this PR locally once.

**Comment by jayantbh** [comment]
Type checking doesn't seem to work via the CLI, **but I suspect it isn't actually related to this PR**.

```
➜  web-server git:(feat/next-upgrade) yarn tsc                                                      
yarn run v1.22.22
$ /Users/jayant/Projects/dora-metrics/web-server/node_modules/.bin/tsc
/Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:116479
      throw e;
      ^

RangeError: Map maximum size exceeded
    at Map.set (<anonymous>)
    at recursiveTypeRelatedTo (/Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:60782:18)
    at isRelatedTo (/Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:60199:122)
    at checkTypeRelatedTo (/Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:59834:20)
    at isTypeRelatedTo (/Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:59778:14)
    at isTypeAssignableTo (/Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:58892:12)
    at typeMaybeAssignableTo (/Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:64446:11)
    at /Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:64463:58
    at filter (/Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:169:23)
    at filterType (/Users/jayant/Projects/dora-metrics/web-server/node_modules/typescript/lib/tsc.js:64752:24)

Node.js v22.10.0
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by jayantbh** [comment]
Okay, the issue I mentioned above is now fixed in main. Please rebase, address the super trivial comment left, and I think we're good.

File: .github/workflows/pre-commit.yml
Comment (Reterics) [reply .github/workflows/pre-commit.yml:1]: Removed contributing.md from this PR, and i rebased on origin/main

**Comment by Reterics** [comment]
> Okay, the issue I mentioned above is now fixed in main. Please rebase, address the super trivial comment left, and I think we're good.

After rebase i verified with "yarn tsc " you mentioned above. It is working. Hopefully it will be fine now.

**Comment by jayantbh** [comment]
Just confirming, did `yarn tsc` in the web-server module work fine for you?

**Comment by Reterics** [comment]
> Just confirming, did `yarn tsc` in the web-server module work fine for you?

Yes it is working fine. (both Ubuntu and Windows 11)
image

image

**Review by jayantbh — APPROVED** [review]

**Comment by jayantbh** [comment]
Great stuff, @Reterics!
Thank you!

**Pull request merged**